packages/flutter/lib/src/widgets/ticker_provider.dart

```diff
@@ -0,0 +1,220 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/scheduler.dart';
+import 'package:meta/meta.dart';
+
+import 'framework.dart';
+
+export 'package:flutter/scheduler.dart' show TickerProvider;
+
+/// Enables or disables tickers (and thus animation controllers) in the widget
+/// subtree.
+///
+/// This only works if [AnimationController] objects are created using
+/// widget-aware ticker providers. For example, using a
+/// [TickerProviderStateMixin] or a [SingleTickerProviderStateMixin].
+class TickerMode extends InheritedWidget {
+  /// Creates a widget that enables or disables tickers.
+  ///
+  /// The [enabled] argument must not be null.
+  TickerMode({
+    Key key,
+    @required this.enabled,
+    Widget child
+  }) : super(key: key, child: child) {
+    assert(enabled != null);
+  }
+
+  /// The current ticker mode of this subtree.
+  ///
+  /// If true, then tickers in this subtree will tick.
+  ///
+  /// If false, then tickers in this subtree will not tick. Animations driven by
+  /// such tickers are not paused, they just don't call their callbacks. Time
+  /// still elapses.
+  final bool enabled;
```

Comment:
Actually calling this "muted" here leads to a weird case where you set things to false if they're on, which is weird. I'm not sure what to do.